// livepeer/dist/models/components/attestation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Attestation = exports.AttestationStorage = exports.AttestationIpfs = exports.SignatureType = exports.Message = exports.Attestations = exports.Domain = exports.Version = exports.Name = exports.PrimaryType = void 0;
var utils_1 = require("../../internal/utils");
var storagestatus_1 = require("./storagestatus");
var class_transformer_1 = require("class-transformer");
/**
 * Video Metadata EIP-712 primaryType
 */
var PrimaryType;
(function (PrimaryType) {
    PrimaryType["VideoAttestation"] = "VideoAttestation";
})(PrimaryType = exports.PrimaryType || (exports.PrimaryType = {}));
var Name;
(function (Name) {
    Name["VerifiableVideo"] = "Verifiable Video";
})(Name = exports.Name || (exports.Name = {}));
var Version;
(function (Version) {
    Version["One"] = "1";
})(Version = exports.Version || (exports.Version = {}));
/**
 * Video Metadata EIP-712 domain
 */
var Domain = /** @class */ (function (_super) {
    __extends(Domain, _super);
    function Domain() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], Domain.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "version" }),
        __metadata("design:type", String)
    ], Domain.prototype, "version", void 0);
    return Domain;
}(utils_1.SpeakeasyBase));
exports.Domain = Domain;
var Attestations = /** @class */ (function (_super) {
    __extends(Attestations, _super);
    function Attestations() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "role" }),
        __metadata("design:type", String)
    ], Attestations.prototype, "role", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "address" }),
        __metadata("design:type", String)
    ], Attestations.prototype, "address", void 0);
    return Attestations;
}(utils_1.SpeakeasyBase));
exports.Attestations = Attestations;
/**
 * Video Metadata EIP-712 message content
 */
var Message = /** @class */ (function (_super) {
    __extends(Message, _super);
    function Message() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "video" }),
        __metadata("design:type", String)
    ], Message.prototype, "video", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: Attestations }),
        (0, class_transformer_1.Expose)({ name: "attestations" }),
        (0, class_transformer_1.Type)(function () { return Attestations; }),
        __metadata("design:type", Array)
    ], Message.prototype, "attestations", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "signer" }),
        __metadata("design:type", String)
    ], Message.prototype, "signer", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "timestamp" }),
        __metadata("design:type", Number)
    ], Message.prototype, "timestamp", void 0);
    return Message;
}(utils_1.SpeakeasyBase));
exports.Message = Message;
var SignatureType;
(function (SignatureType) {
    SignatureType["Eip712"] = "eip712";
    SignatureType["Flow"] = "flow";
})(SignatureType = exports.SignatureType || (exports.SignatureType = {}));
var AttestationIpfs = /** @class */ (function (_super) {
    __extends(AttestationIpfs, _super);
    function AttestationIpfs() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "cid" }),
        __metadata("design:type", String)
    ], AttestationIpfs.prototype, "cid", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "url" }),
        __metadata("design:type", String)
    ], AttestationIpfs.prototype, "url", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "gatewayUrl" }),
        __metadata("design:type", String)
    ], AttestationIpfs.prototype, "gatewayUrl", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "updatedAt" }),
        __metadata("design:type", Number)
    ], AttestationIpfs.prototype, "updatedAt", void 0);
    return AttestationIpfs;
}(utils_1.SpeakeasyBase));
exports.AttestationIpfs = AttestationIpfs;
var AttestationStorage = /** @class */ (function (_super) {
    __extends(AttestationStorage, _super);
    function AttestationStorage() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "ipfs" }),
        (0, class_transformer_1.Type)(function () { return AttestationIpfs; }),
        __metadata("design:type", AttestationIpfs)
    ], AttestationStorage.prototype, "ipfs", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        (0, class_transformer_1.Type)(function () { return storagestatus_1.StorageStatus; }),
        __metadata("design:type", storagestatus_1.StorageStatus)
    ], AttestationStorage.prototype, "status", void 0);
    return AttestationStorage;
}(utils_1.SpeakeasyBase));
exports.AttestationStorage = AttestationStorage;
var Attestation = /** @class */ (function (_super) {
    __extends(Attestation, _super);
    function Attestation() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], Attestation.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "primaryType" }),
        __metadata("design:type", String)
    ], Attestation.prototype, "primaryType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "domain" }),
        (0, class_transformer_1.Type)(function () { return Domain; }),
        __metadata("design:type", Domain)
    ], Attestation.prototype, "domain", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "message" }),
        (0, class_transformer_1.Type)(function () { return Message; }),
        __metadata("design:type", Message)
    ], Attestation.prototype, "message", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "signature" }),
        __metadata("design:type", String)
    ], Attestation.prototype, "signature", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        __metadata("design:type", Number)
    ], Attestation.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "signatureType" }),
        __metadata("design:type", String)
    ], Attestation.prototype, "signatureType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "storage" }),
        (0, class_transformer_1.Type)(function () { return AttestationStorage; }),
        __metadata("design:type", AttestationStorage)
    ], Attestation.prototype, "storage", void 0);
    return Attestation;
}(utils_1.SpeakeasyBase));
exports.Attestation = Attestation;
