// livepeer/dist/models/components/ffmpegprofile.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FfmpegProfile = exports.Encoder = exports.Profile = void 0;
var utils_1 = require("../../internal/utils");
var class_transformer_1 = require("class-transformer");
var Profile;
(function (Profile) {
    Profile["H264Baseline"] = "H264Baseline";
    Profile["H264Main"] = "H264Main";
    Profile["H264High"] = "H264High";
    Profile["H264ConstrainedHigh"] = "H264ConstrainedHigh";
})(Profile = exports.Profile || (exports.Profile = {}));
var Encoder;
(function (Encoder) {
    Encoder["H264"] = "h264";
    Encoder["Hevc"] = "hevc";
    Encoder["Vp8"] = "vp8";
    Encoder["Vp9"] = "vp9";
})(Encoder = exports.Encoder || (exports.Encoder = {}));
/**
 * LMPS ffmpeg profile
 */
var FfmpegProfile = /** @class */ (function (_super) {
    __extends(FfmpegProfile, _super);
    function FfmpegProfile() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "width" }),
        __metadata("design:type", Number)
    ], FfmpegProfile.prototype, "width", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], FfmpegProfile.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "height" }),
        __metadata("design:type", Number)
    ], FfmpegProfile.prototype, "height", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "bitrate" }),
        __metadata("design:type", Number)
    ], FfmpegProfile.prototype, "bitrate", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "fps" }),
        __metadata("design:type", Number)
    ], FfmpegProfile.prototype, "fps", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "fpsDen" }),
        __metadata("design:type", Number)
    ], FfmpegProfile.prototype, "fpsDen", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "gop" }),
        __metadata("design:type", String)
    ], FfmpegProfile.prototype, "gop", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "profile" }),
        __metadata("design:type", String)
    ], FfmpegProfile.prototype, "profile", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "encoder" }),
        __metadata("design:type", String)
    ], FfmpegProfile.prototype, "encoder", void 0);
    return FfmpegProfile;
}(utils_1.SpeakeasyBase));
exports.FfmpegProfile = FfmpegProfile;
