// livepeer/dist/models/components/webhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.Status = exports.LastFailure = exports.Events = void 0;
var utils_1 = require("../../internal/utils");
var class_transformer_1 = require("class-transformer");
var Events;
(function (Events) {
    Events["StreamStarted"] = "stream.started";
    Events["StreamDetection"] = "stream.detection";
    Events["StreamIdle"] = "stream.idle";
    Events["RecordingReady"] = "recording.ready";
    Events["RecordingStarted"] = "recording.started";
    Events["RecordingWaiting"] = "recording.waiting";
    Events["MultistreamConnected"] = "multistream.connected";
    Events["MultistreamError"] = "multistream.error";
    Events["MultistreamDisconnected"] = "multistream.disconnected";
    Events["PlaybackUserNew"] = "playback.user.new";
    Events["PlaybackAccessControl"] = "playback.accessControl";
    Events["AssetCreated"] = "asset.created";
    Events["AssetUpdated"] = "asset.updated";
    Events["AssetFailed"] = "asset.failed";
    Events["AssetReady"] = "asset.ready";
    Events["AssetDeleted"] = "asset.deleted";
    Events["TaskSpawned"] = "task.spawned";
    Events["TaskUpdated"] = "task.updated";
    Events["TaskCompleted"] = "task.completed";
    Events["TaskFailed"] = "task.failed";
})(Events = exports.Events || (exports.Events = {}));
/**
 * failure timestamp and error message with status code
 */
var LastFailure = /** @class */ (function (_super) {
    __extends(LastFailure, _super);
    function LastFailure() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "timestamp" }),
        __metadata("design:type", Number)
    ], LastFailure.prototype, "timestamp", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "error" }),
        __metadata("design:type", String)
    ], LastFailure.prototype, "error", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "response" }),
        __metadata("design:type", String)
    ], LastFailure.prototype, "response", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "statusCode" }),
        __metadata("design:type", Number)
    ], LastFailure.prototype, "statusCode", void 0);
    return LastFailure;
}(utils_1.SpeakeasyBase));
exports.LastFailure = LastFailure;
/**
 * status of webhook
 */
var Status = /** @class */ (function (_super) {
    __extends(Status, _super);
    function Status() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "lastFailure" }),
        (0, class_transformer_1.Type)(function () { return LastFailure; }),
        __metadata("design:type", LastFailure)
    ], Status.prototype, "lastFailure", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "lastTriggeredAt" }),
        __metadata("design:type", Number)
    ], Status.prototype, "lastTriggeredAt", void 0);
    return Status;
}(utils_1.SpeakeasyBase));
exports.Status = Status;
var Webhook = /** @class */ (function (_super) {
    __extends(Webhook, _super);
    function Webhook() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], Webhook.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], Webhook.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        __metadata("design:type", Number)
    ], Webhook.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "events" }),
        __metadata("design:type", Array)
    ], Webhook.prototype, "events", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "url" }),
        __metadata("design:type", String)
    ], Webhook.prototype, "url", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "streamId" }),
        __metadata("design:type", String)
    ], Webhook.prototype, "streamId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        (0, class_transformer_1.Type)(function () { return Status; }),
        __metadata("design:type", Status)
    ], Webhook.prototype, "status", void 0);
    return Webhook;
}(utils_1.SpeakeasyBase));
exports.Webhook = Webhook;
