// livepeer/dist/sdk/session.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Session = void 0;
var utils = __importStar(require("../internal/utils"));
var components = __importStar(require("../models/components"));
var errors = __importStar(require("../models/errors"));
var operations = __importStar(require("../models/operations"));
var Session = /** @class */ (function () {
    function Session(sdkConfig) {
        this.sdkConfiguration = sdkConfig;
    }
    /**
     * Retrieve sessions
     */
    Session.prototype.getAll = function (config) {
        var _a, _b;
        return __awaiter(this, void 0, void 0, function () {
            var baseURL, operationUrl, client, globalSecurity, properties, headers, httpRes, responseContentType, res, decodedRes, resFieldDepth;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        baseURL = utils.templateUrl(this.sdkConfiguration.serverURL, this.sdkConfiguration.serverDefaults);
                        operationUrl = baseURL.replace(/\/$/, "") + "/session";
                        client = this.sdkConfiguration.defaultClient;
                        globalSecurity = this.sdkConfiguration.security;
                        if (!(typeof globalSecurity === "function")) return [3 /*break*/, 2];
                        return [4 /*yield*/, globalSecurity()];
                    case 1:
                        globalSecurity = _c.sent();
                        _c.label = 2;
                    case 2:
                        if (!(globalSecurity instanceof utils.SpeakeasyBase)) {
                            globalSecurity = new components.Security(globalSecurity);
                        }
                        properties = utils.parseSecurityProperties(globalSecurity);
                        headers = __assign(__assign({}, config === null || config === void 0 ? void 0 : config.headers), properties.headers);
                        headers["Accept"] = "application/json";
                        headers["user-agent"] = this.sdkConfiguration.userAgent;
                        return [4 /*yield*/, client.request(__assign({ validateStatus: function () { return true; }, url: operationUrl, method: "get", headers: headers, responseType: "arraybuffer" }, config))];
                    case 3:
                        httpRes = _c.sent();
                        responseContentType = (_b = (_a = httpRes === null || httpRes === void 0 ? void 0 : httpRes.headers) === null || _a === void 0 ? void 0 : _a["content-type"]) !== null && _b !== void 0 ? _b : "";
                        if ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == null) {
                            throw new Error("status code not found in response: ".concat(httpRes));
                        }
                        res = new operations.GetSessionsResponse({
                            statusCode: httpRes.status,
                            contentType: responseContentType,
                            rawResponse: httpRes,
                        });
                        decodedRes = new TextDecoder().decode(httpRes === null || httpRes === void 0 ? void 0 : httpRes.data);
                        switch (true) {
                            case (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == 200:
                                if (utils.matchContentType(responseContentType, "application/json")) {
                                    res.data = [];
                                    resFieldDepth = utils.getResFieldDepth(res);
                                    res.data = utils.objectToClass(JSON.parse(decodedRes), components.Session, resFieldDepth);
                                }
                                else {
                                    throw new errors.SDKError("unknown content-type received: " + responseContentType, httpRes.status, decodedRes, httpRes);
                                }
                                break;
                            case ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 400 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 500) ||
                                ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 500 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 600):
                                throw new errors.SDKError("API error occurred", httpRes.status, decodedRes, httpRes);
                        }
                        return [2 /*return*/, res];
                }
            });
        });
    };
    /**
     * Retrieve a session
     */
    Session.prototype.get = function (id, config) {
        var _a, _b;
        return __awaiter(this, void 0, void 0, function () {
            var req, baseURL, operationUrl, client, globalSecurity, properties, headers, httpRes, responseContentType, res, decodedRes;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        req = new operations.GetSessionRequest({
                            id: id,
                        });
                        baseURL = utils.templateUrl(this.sdkConfiguration.serverURL, this.sdkConfiguration.serverDefaults);
                        operationUrl = utils.generateURL(baseURL, "/session/{id}", req);
                        client = this.sdkConfiguration.defaultClient;
                        globalSecurity = this.sdkConfiguration.security;
                        if (!(typeof globalSecurity === "function")) return [3 /*break*/, 2];
                        return [4 /*yield*/, globalSecurity()];
                    case 1:
                        globalSecurity = _c.sent();
                        _c.label = 2;
                    case 2:
                        if (!(globalSecurity instanceof utils.SpeakeasyBase)) {
                            globalSecurity = new components.Security(globalSecurity);
                        }
                        properties = utils.parseSecurityProperties(globalSecurity);
                        headers = __assign(__assign({}, config === null || config === void 0 ? void 0 : config.headers), properties.headers);
                        headers["Accept"] = "application/json";
                        headers["user-agent"] = this.sdkConfiguration.userAgent;
                        return [4 /*yield*/, client.request(__assign({ validateStatus: function () { return true; }, url: operationUrl, method: "get", headers: headers, responseType: "arraybuffer" }, config))];
                    case 3:
                        httpRes = _c.sent();
                        responseContentType = (_b = (_a = httpRes === null || httpRes === void 0 ? void 0 : httpRes.headers) === null || _a === void 0 ? void 0 : _a["content-type"]) !== null && _b !== void 0 ? _b : "";
                        if ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == null) {
                            throw new Error("status code not found in response: ".concat(httpRes));
                        }
                        res = new operations.GetSessionResponse({
                            statusCode: httpRes.status,
                            contentType: responseContentType,
                            rawResponse: httpRes,
                        });
                        decodedRes = new TextDecoder().decode(httpRes === null || httpRes === void 0 ? void 0 : httpRes.data);
                        switch (true) {
                            case (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == 200:
                                if (utils.matchContentType(responseContentType, "application/json")) {
                                    res.session = utils.objectToClass(JSON.parse(decodedRes), components.Session);
                                }
                                else {
                                    throw new errors.SDKError("unknown content-type received: " + responseContentType, httpRes.status, decodedRes, httpRes);
                                }
                                break;
                            case ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 400 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 500) ||
                                ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 500 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 600):
                                throw new errors.SDKError("API error occurred", httpRes.status, decodedRes, httpRes);
                        }
                        return [2 /*return*/, res];
                }
            });
        });
    };
    /**
     * Retrieve Recorded Sessions
     */
    Session.prototype.getRecorded = function (parentId, record, config) {
        var _a, _b;
        return __awaiter(this, void 0, void 0, function () {
            var req, baseURL, operationUrl, client, globalSecurity, properties, headers, queryParams, httpRes, responseContentType, res, decodedRes, resFieldDepth;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        req = new operations.GetRecordedSessionsRequest({
                            parentId: parentId,
                            record: record,
                        });
                        baseURL = utils.templateUrl(this.sdkConfiguration.serverURL, this.sdkConfiguration.serverDefaults);
                        operationUrl = utils.generateURL(baseURL, "/stream/{parentId}/sessions", req);
                        client = this.sdkConfiguration.defaultClient;
                        globalSecurity = this.sdkConfiguration.security;
                        if (!(typeof globalSecurity === "function")) return [3 /*break*/, 2];
                        return [4 /*yield*/, globalSecurity()];
                    case 1:
                        globalSecurity = _c.sent();
                        _c.label = 2;
                    case 2:
                        if (!(globalSecurity instanceof utils.SpeakeasyBase)) {
                            globalSecurity = new components.Security(globalSecurity);
                        }
                        properties = utils.parseSecurityProperties(globalSecurity);
                        headers = __assign(__assign({}, config === null || config === void 0 ? void 0 : config.headers), properties.headers);
                        queryParams = utils.serializeQueryParams(req);
                        headers["Accept"] = "application/json";
                        headers["user-agent"] = this.sdkConfiguration.userAgent;
                        return [4 /*yield*/, client.request(__assign({ validateStatus: function () { return true; }, url: operationUrl + queryParams, method: "get", headers: headers, responseType: "arraybuffer" }, config))];
                    case 3:
                        httpRes = _c.sent();
                        responseContentType = (_b = (_a = httpRes === null || httpRes === void 0 ? void 0 : httpRes.headers) === null || _a === void 0 ? void 0 : _a["content-type"]) !== null && _b !== void 0 ? _b : "";
                        if ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == null) {
                            throw new Error("status code not found in response: ".concat(httpRes));
                        }
                        res = new operations.GetRecordedSessionsResponse({
                            statusCode: httpRes.status,
                            contentType: responseContentType,
                            rawResponse: httpRes,
                        });
                        decodedRes = new TextDecoder().decode(httpRes === null || httpRes === void 0 ? void 0 : httpRes.data);
                        switch (true) {
                            case (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == 200:
                                if (utils.matchContentType(responseContentType, "application/json")) {
                                    res.data = [];
                                    resFieldDepth = utils.getResFieldDepth(res);
                                    res.data = utils.objectToClass(JSON.parse(decodedRes), components.Session, resFieldDepth);
                                }
                                else {
                                    throw new errors.SDKError("unknown content-type received: " + responseContentType, httpRes.status, decodedRes, httpRes);
                                }
                                break;
                            case ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 400 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 500) ||
                                ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 500 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 600):
                                throw new errors.SDKError("API error occurred", httpRes.status, decodedRes, httpRes);
                        }
                        return [2 /*return*/, res];
                }
            });
        });
    };
    /**
     * Retrieve clips of a session
     */
    Session.prototype.getAllClips = function (id, config) {
        var _a, _b;
        return __awaiter(this, void 0, void 0, function () {
            var req, baseURL, operationUrl, client, globalSecurity, properties, headers, httpRes, responseContentType, res, decodedRes, resFieldDepth;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        req = new operations.GetSessionIdClipsRequest({
                            id: id,
                        });
                        baseURL = utils.templateUrl(this.sdkConfiguration.serverURL, this.sdkConfiguration.serverDefaults);
                        operationUrl = utils.generateURL(baseURL, "/session/{id}/clips", req);
                        client = this.sdkConfiguration.defaultClient;
                        globalSecurity = this.sdkConfiguration.security;
                        if (!(typeof globalSecurity === "function")) return [3 /*break*/, 2];
                        return [4 /*yield*/, globalSecurity()];
                    case 1:
                        globalSecurity = _c.sent();
                        _c.label = 2;
                    case 2:
                        if (!(globalSecurity instanceof utils.SpeakeasyBase)) {
                            globalSecurity = new components.Security(globalSecurity);
                        }
                        properties = utils.parseSecurityProperties(globalSecurity);
                        headers = __assign(__assign({}, config === null || config === void 0 ? void 0 : config.headers), properties.headers);
                        headers["Accept"] = "application/json";
                        headers["user-agent"] = this.sdkConfiguration.userAgent;
                        return [4 /*yield*/, client.request(__assign({ validateStatus: function () { return true; }, url: operationUrl, method: "get", headers: headers, responseType: "arraybuffer" }, config))];
                    case 3:
                        httpRes = _c.sent();
                        responseContentType = (_b = (_a = httpRes === null || httpRes === void 0 ? void 0 : httpRes.headers) === null || _a === void 0 ? void 0 : _a["content-type"]) !== null && _b !== void 0 ? _b : "";
                        if ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == null) {
                            throw new Error("status code not found in response: ".concat(httpRes));
                        }
                        res = new operations.GetSessionIdClipsResponse({
                            statusCode: httpRes.status,
                            contentType: responseContentType,
                            rawResponse: httpRes,
                        });
                        decodedRes = new TextDecoder().decode(httpRes === null || httpRes === void 0 ? void 0 : httpRes.data);
                        switch (true) {
                            case (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) == 200:
                                if (utils.matchContentType(responseContentType, "application/json")) {
                                    res.data = [];
                                    resFieldDepth = utils.getResFieldDepth(res);
                                    res.data = utils.objectToClass(JSON.parse(decodedRes), components.Asset, resFieldDepth);
                                }
                                else {
                                    throw new errors.SDKError("unknown content-type received: " + responseContentType, httpRes.status, decodedRes, httpRes);
                                }
                                break;
                            case ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 400 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 500) ||
                                ((httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) >= 500 && (httpRes === null || httpRes === void 0 ? void 0 : httpRes.status) < 600):
                                throw new errors.SDKError("API error occurred", httpRes.status, decodedRes, httpRes);
                        }
                        return [2 /*return*/, res];
                }
            });
        });
    };
    return Session;
}());
exports.Session = Session;
