// livepeer/dist/models/operations/getcreatormetrics.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCreatorMetricsResponse = exports.GetCreatorMetricsRequest = exports.QueryParamBreakdownBy = exports.QueryParamTimeStep = void 0;
var utils_1 = require("../../internal/utils");
var components = __importStar(require("../../models/components"));
/**
 * The time step to aggregate viewership metrics by
 */
var QueryParamTimeStep;
(function (QueryParamTimeStep) {
    QueryParamTimeStep["Hour"] = "hour";
    QueryParamTimeStep["Day"] = "day";
    QueryParamTimeStep["Week"] = "week";
    QueryParamTimeStep["Month"] = "month";
    QueryParamTimeStep["Year"] = "year";
})(QueryParamTimeStep = exports.QueryParamTimeStep || (exports.QueryParamTimeStep = {}));
var QueryParamBreakdownBy;
(function (QueryParamBreakdownBy) {
    QueryParamBreakdownBy["DeviceType"] = "deviceType";
    QueryParamBreakdownBy["Device"] = "device";
    QueryParamBreakdownBy["Cpu"] = "cpu";
    QueryParamBreakdownBy["Os"] = "os";
    QueryParamBreakdownBy["Browser"] = "browser";
    QueryParamBreakdownBy["BrowserEngine"] = "browserEngine";
    QueryParamBreakdownBy["Continent"] = "continent";
    QueryParamBreakdownBy["Country"] = "country";
    QueryParamBreakdownBy["Subdivision"] = "subdivision";
    QueryParamBreakdownBy["Timezone"] = "timezone";
    QueryParamBreakdownBy["ViewerId"] = "viewerId";
})(QueryParamBreakdownBy = exports.QueryParamBreakdownBy || (exports.QueryParamBreakdownBy = {}));
var GetCreatorMetricsRequest = /** @class */ (function (_super) {
    __extends(GetCreatorMetricsRequest, _super);
    function GetCreatorMetricsRequest() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=from" }),
        __metadata("design:type", Object)
    ], GetCreatorMetricsRequest.prototype, "from", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=to" }),
        __metadata("design:type", Object)
    ], GetCreatorMetricsRequest.prototype, "to", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=timeStep" }),
        __metadata("design:type", String)
    ], GetCreatorMetricsRequest.prototype, "timeStep", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=assetId" }),
        __metadata("design:type", String)
    ], GetCreatorMetricsRequest.prototype, "assetId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=streamId" }),
        __metadata("design:type", String)
    ], GetCreatorMetricsRequest.prototype, "streamId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=creatorId" }),
        __metadata("design:type", String)
    ], GetCreatorMetricsRequest.prototype, "creatorId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=breakdownBy[]" }),
        __metadata("design:type", Array)
    ], GetCreatorMetricsRequest.prototype, "breakdownBy", void 0);
    return GetCreatorMetricsRequest;
}(utils_1.SpeakeasyBase));
exports.GetCreatorMetricsRequest = GetCreatorMetricsRequest;
var GetCreatorMetricsResponse = /** @class */ (function (_super) {
    __extends(GetCreatorMetricsResponse, _super);
    function GetCreatorMetricsResponse() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", String)
    ], GetCreatorMetricsResponse.prototype, "contentType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Number)
    ], GetCreatorMetricsResponse.prototype, "statusCode", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Object)
    ], GetCreatorMetricsResponse.prototype, "rawResponse", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: components.ViewershipMetric }),
        __metadata("design:type", Array)
    ], GetCreatorMetricsResponse.prototype, "data", void 0);
    return GetCreatorMetricsResponse;
}(utils_1.SpeakeasyBase));
exports.GetCreatorMetricsResponse = GetCreatorMetricsResponse;
